src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -472,9 +249,60 @@ private Future<Void> storeUpdatedPieceRecord(Piece piece, RequestContext request
       });
   }
 
-
-  protected List<String> getPoLineIdsForUpdatedPieces(Map<String, List<Piece>> piecesGroupedByPoLine) {
-    return EntryStream
+  /**
+   * @param pieceId piece id
+   * @return PO Line id corresponding to passed pieceId from request data
+   */
+  private String getPoLineIdByPieceId(String pieceId) {
+    return StreamEx
+      .ofKeys(piecesByLineId, values -> values.containsKey(pieceId))
+      .findFirst()
+      .orElse(EMPTY);
+  }
+
+  //-------------------------------------------------------------------------------------
+  /*
+  updateOrderAndPoLinesStatus
+   */
+
+  protected Future<Map<String, List<Piece>>> updateOrderAndPoLinesStatus(Map<String, List<Piece>> piecesGroupedByPoLine,
+                                                                         RequestContext requestContext,
+                                                                         Consumer<List<PoLine>> updateOrderStatus) {
+    if (MapUtils.isEmpty(piecesGroupedByPoLine)) {
+      return Future.succeededFuture(piecesGroupedByPoLine);
+    }
+    List<String> poLineIdsForUpdatedPieces = getPoLineIdsForUpdatedPieces(piecesGroupedByPoLine);
+    // Once all PO Lines are retrieved from storage check if receipt status
+    // requires update and persist in storage
+    return getPoLines(poLineIdsForUpdatedPieces, requestContext).compose(poLines -> {
+        // Calculate expected status for each PO Line and update with new one if required
+        // Skip status update if PO line status is Ongoing
+        List<Future<String>> futures = new ArrayList<>();
```

Comment:
we created futures but we are not using it, can we simpplfy this parts

Reply:
The are used at line 279: `collectResultsOnSuccess(futures)`